applications/DEMApplication/custom_constitutive/DEM_compound_constitutive_law_for_PBM.h

```diff
@@ -0,0 +1,56 @@
+#ifndef KRATOS_DEM_COMPOUND_CONSTITUTIVE_LAW_FOR_PBM_H
+#define KRATOS_DEM_COMPOUND_CONSTITUTIVE_LAW_FOR_PBM_H
+
+#include "includes/define.h"
+#include "includes/serializer.h"
+
+namespace Kratos{
+
+template <class BondCL, class UnbondCL>
+
+class KRATOS_API(DEM_APPLICATION) DEM_compound_constitutive_law_for_PBM : public BondCL {
+
+public:
+
+    KRATOS_CLASS_POINTER_DEFINITION(DEM_compound_constitutive_law_for_PBM);
+
+    DEMContinuumConstitutiveLaw::Pointer Clone() const override {
+        DEMContinuumConstitutiveLaw::Pointer p_clone(new DEM_compound_constitutive_law_for_PBM<BondCL, UnbondCL>(*this));
+        return p_clone;
+    }
+
+    std::unique_ptr<DEMContinuumConstitutiveLaw> CloneUnique() {
+        return std::unique_ptr<DEMContinuumConstitutiveLaw>{new DEM_compound_constitutive_law_for_PBM<BondCL, UnbondCL>(*this)};
+    }
+
+    virtual ~DEM_compound_constitutive_law_for_PBM() {}
+
+    //add
+    double ComputeNormalUnbondedForce(double indentation) override {
+        return mCCL.CalculateNormalForce(indentation);
+    }
+
+    void InitializeContact(SphericParticle* const element1, SphericParticle* const element2, const double indentation) override {
+        return mCCL.InitializeContact(element1, element2, indentation);
+    }
+
+private:
+
+    UnbondCL mCCL;
```

Comment:
This could be a pointer instead of a full object. We can discuss it.

Reply:
Thanks! Discontinuum laws do not save memory because they do not have memory variables, so it is similar to save a point or an instance. Also: what would be the pointer pointing at?

Reply:
Right, if we use a pointer the structure will be harder to understand, and the memory overhead is small or zero.